src/instructlab/rag/convert.py

```diff
@@ -0,0 +1,250 @@
+            break
+
+    if sdg_models_path is not None:
+        try:
+            with open(
+                os.path.join(sdg_models_path, "config.yaml"), "r", encoding="utf-8"
+            ) as file:
+                config = yaml.safe_load(file)
+                docling_model_path = config["models"][0]["path"]
+        except (FileNotFoundError, NotADirectoryError, PermissionError) as e:
+            logger.warning(f"unable to read docling models path from config.yaml {e}")
+
+    pipeline_options = PdfPipelineOptions(
+        artifacts_path=docling_model_path,
+        do_ocr=False,
+    )
+    ocr_options = _resolve_ocr_options()
+    if ocr_options is not None:
+        pipeline_options.do_ocr = True
+        pipeline_options.ocr_options = ocr_options
+
+    _load_converter_and_format_options()
+    doc_converter = DocumentConverter(  # pylint: disable=E0602
+        format_options={
+            InputFormat.PDF: PdfFormatOption(pipeline_options=pipeline_options)  # pylint: disable=E0602
+        }
+    )
+
+    return doc_converter
+
+
+# Adapted from sdg.utils.chunkers because that code is being refactored so we want to avoid importing anything from it.
+# TODO: Once the code base has settled down, we should make sure this code exists only in one place.
+def _resolve_ocr_options() -> OcrOptions:
+    """
+    Attempts to resolve OCR options for a PDF document. It first tries to use the Tesseract OCR library,
+    and if that fails, it tries the EasyOCR library. If neither of these libraries are available, it
+    returns None to indicate that OCR will not be used. Note that it imports the OCR libraries inside
+    the code if/when they are needed because they are kind of heavy.
+    """
+    # First, attempt to use tesserocr
+    try:
+        ocr_options = TesseractOcrOptions()
+
+        # pylint: disable=import-outside-toplevel
+        # Third Party
+        from docling.models.tesseract_ocr_model import (  # type: ignore[import-untyped]
+            TesseractOcrModel,
+        )
+
+        _ = TesseractOcrModel(True, ocr_options)
+        return ocr_options
+    except ImportError:
+        # No tesserocr, so try something else
+        pass
+    try:
+        # pylint: disable=import-outside-toplevel
+        # Third Party
+        from docling.models.easyocr_model import (  # type: ignore[import-untyped]
+            EasyOcrModel,
+        )
+
+        ocr_options = EasyOcrOptions()
+
+        # Keep easyocr models on the CPU instead of GPU
+        ocr_options.use_gpu = False
+        # triggers torch loading, import lazily
+
+        _ = EasyOcrModel(True, ocr_options)
+        return ocr_options
+    except ImportError:
+        # no easyocr either, so don't use any OCR
+        logger.error(
+            "Failed to load Tesseract and EasyOCR - disabling optical character recognition in PDF documents"
+        )
+        return None
+
+
+def _load_converter_and_format_options():
```

Comment:
what is this function actually doing? just imports and global initializations?

Reply:
Yes, so that leanimports.py can exclude torch.   See the doc string I added to leanimports.py for more context.  I can add more commenting here too if that would help.